src/Compilers/CSharp/Portable/Syntax/DirectiveTriviaSyntax.cs

```diff
@@ -166,14 +166,15 @@ private DirectiveTriviaSyntax GetNextRelatedDirective()
                 case SyntaxKind.ElifDirectiveTrivia:
                     while (d != null)
                     {
+                        d = d.GetNextPossiblyRelatedDirective();
```

Comment:
Moving this assign up here means that the following d.Kind() call can throw null ref exception.